text/0000-ubi-based-stacks.md

```diff
@@ -0,0 +1,181 @@
+native compilation are complex, and may require the Builder to extend an appropriate base image to provide 
+the require capabilities in a supported manner. 
+
+As additional runtimes are added to the initial image over time, it may make sense to follow the same 
+general approach as the Ubuntu stacks have, and aim for a small but manageable number of stacks that meet definable 
+use cases. (eg, 'one size fits all', 'just the most used runtimes')
+
+
+## Rationale and Alternatives
+
+With the goal in mind of creating a Stack based on UBI using system packaged depdencies, these alternatives 
+have been considered...
+
+#### Builders per Runtime
+This option would have 1 Builder image per runtime, paired with a matching run image per runtime, with the runtime
+depdencies baked into the builder/run images. This option did not require Extensions from upstream, but would 
+result in a large collection of single purpose stacks which rapidly becomes unmanageable as you consider versioning
+across runtimes (eg, there would likely need to be java 8,11,17 stacks). Exisiting Paketo buildpacks would be 
+checked to ensure they would honor & use a dependency already in the builder image, over one the buildpack would
+normally download & use.
+
+In addition this transfers the responsibility of a lot of the 'magic' of Buildpacks to the end user, requiring them 
+to know in advance which builder image to use with a given project, and to submit to the correct build pipeline, or
+adapt the pipeline per project. 
+
+#### 'Phat' Builder Image
+This option would have 1 Builder image with many runtimes baked in, and then a selection of run images with
+appropriate runtimes baked in that would be selected by the Builder via Extensions. As with the previous option, 
+existing buildpacks would be checked to ensure they would use the baked in dependencies as a priority. 
+
+This approach leads to a very large Builder image, which is suboptimal from an end user perspective, plus, as all 
+runtimes are baked in, there would likely be frequent updates to the Builder image to incorporate fixes & security
+updates to each contained runtime, leading to the end-user requiring a fresh version of the builder image for almost 
+every update. A large image + frequent updates creates a less than ideal development experience.
+
+#### UBI without rpms
+This option would just take the existing Paketo buildpacks as-is and run them on a UBI base image. As with all the 
+options here, testing would be required for all contained buildpacks to ensure any system or shell interactions were 
+unaffected by the Ubuntu->UBI switch. 
+
+Although this approach would produce functional application images, the resulting images would not be using UBI 
+installed runtimes, and thus could be unable to qualify for full support in production. This is considered important 
+enough to end users that this option was discarded.
```

Comment:
Just curious, but what if you used the buildpacks to perform the installation but they installed UBI runtime files? Hypothetically speaking, what if the buildpack installed the files from a UBI RPM into a UBI image, just in a different location (since the buildpacks can only write to `/layers`)?

I'm not really proposing this as an alternative, but I'm curious about what is required for a user to be officially supported with UBI. Thanks in advance.

Reply:
Heh.. it's not entirely hypothetical, we looked pretty extensively into if we could install files from various runtime RPMs into `/layers` .. but sadly it's not that simple, as rpm's are intended to install to the system, and we discovered hardcoded path dependencies within some that prevented their install to different locations. From a technical perspective, although this could be corrected, you'd basically be expanding the support offering to allow the installation to non-default paths.. that would likely come with a pile of required testing etc and isn't something that's easy to justify just for buildpacks. 

From my understanding of "supported" you generally today use the precanned ubi8 runtime images, where it's all already installed.. or it's also acceptable to install via rpm. However, I'm just a developer, and my answer shouldn't be taken as a legal response ;) I suspect the actual definition of supported is thrashed out by any support contract by lawyer type people who are far more precise in their definition =)